File: internal/content/testdata/bench/go/lexicographic-order-enumeration.go
func lexical_order(n int) []int {
	result := []int{}
	cur := 1
	for len(result) < n {
		result = append(result, cur)
		if cur*10 <= n {
			cur *= 10
		} else {
			for cur%10 == 9 || cur >= n {
				cur /= 10
			}
			cur++
		}
	}
	return result
}

File: internal/content/testdata/bench/java/lexicographic-order-enumeration.java
import java.util.*;

class Solution {
    public long[] lexical_order(long n) {
        List<Long> result = new ArrayList<>();
        long cur = 1;
        while (result.size() < (int) n) {
            result.add(cur);
            if (cur * 10 <= n) {
                cur *= 10;
            } else {
                while (cur % 10 == 9 || cur >= n) {
                    cur /= 10;
                }
                cur += 1;
            }
        }
        long[] arr = new long[(int) n];
        for (int i = 0; i < result.size(); i++) arr[i] = result.get(i);
        return arr;
    }
}
